### ISSUE TITLE
LDAP Connection

### ISSUE THREAD

**Issue by Laayem** [passwordcockpit/passwordcockpit#6]
Hi, 

I can't connect users when ldap backend is configured, all I've got is the red pop-up (Error, Wrong username or password).
These are the parameters that I added to docker-compose.yml : 
`
PASSWORDCOCKPIT_AUTHENTICATION_TYPE: ldap
PASSWORDCOCKPIT_LDAP_HOST: ip.address.of.dc
PASSWORDCOCKPIT_LDAP_PORT: 389
PASSWORDCOCKPIT_LDAP_USERNAME: uid-name,CN=user,OU=OU,DC=domain,DC=com
PASSWORDCOCKPIT_LDAP_PASSWORD: my_password
PASSWORDCOCKPIT_LDAP_BASEDN: OU=OU,DC=domain,DC=com
PASSWORDCOCKPIT_LDAP_ACCOUNTFILTERFORMAT: (&(memberOf=CN=group,OU=OU,DC=domain,DC=com)(uid=%s))
PASSWORDCOCKPIT_LDAP_BINDREQUIRESDN: 'false' (I tested both without luck)
`
Do you have any hint to help me configure ldap connection?

Does the default admin work if ldap is configured? Which user is the admin?

Thank you in advance.
Mathieu

**Comment by bu3tt1** [comment]
I noticed that PASSWORDCOCKPIT_LDAP_USERNAME is `uid-name,CN=user,OU=OU,DC=domain,DC=com`, correct should be: `uid=name,CN=user,OU=OU,DC=domain,DC=com`

If LDAP is enabled it works for all users, you cannot deactivate it for one user only (admin). The admin must exist in the baseDN.

Username (uid) of the admin is `admin`.

You can add another admin user directly through INSERT in the DB, e.g.: `INSERT INTO user VALUES('1', 'youradminuid', '$2y$10$NpW4tA8NVA68HDXO.p.lFutTbfD1G5Lw9GeR0S3sBuAY3TQ9t4FUO', 'Your', 'Admin', 'it', null, 'admin@your-domain.ch', '1' ,'0');`

**Comment by Laayem** [comment]
I've change uid-name to uid=name but I haven't seen a change.
Our admin is disabled and has been renamed for security reason, I can't create an other one. 

I haven't see logs to check if the connection with ActiveDirectory is OK. Is there any?

**Comment by bu3tt1** [comment]
Have you tried querying and accessing via an LDAP browser to check your PASSWORDCOCKPIT_LDAP_ACCOUNTFILTERFORMAT and PASSWORDCOCKPIT_LDAP_USERNAME?

The user must belong to the group that returns PASSWORDCOCKPIT_LDAP_ACCOUNTFILTERFORMAT. Is it confirmed in your case?

Unfortunately there are no logs for the connection with ActiveDirectory. The error logs can be visible through: `docker logs [container_id]`

At the moment it is not even possible to define LDAP attributes, which are as follows:
- username: uid
- name: givenname
- surname: sn
- email: email

Perhaps your LDAP server does not identify users with these attributes.

Have you tried passwordcockpit without LDAP? Does it work?

**Comment by Laayem** [comment]
PasswordCockpit work without LDAP, but we tried to always work with LDAP to limit number of account for our users. For a password manager, it's mandatory.

After some querying, I've found that we do not use uid or email field, name or surname are not usable because some our users have same firstname or lastname. Usually, we use samAccountName or mail field to identify users. 
If it's not possible, I will continue my research for a new password manager. 

Thanks for your time.
